### PR TITLE
refactor(icon): remove box

### PR DESCRIPTION
**Pull request by baptisteArno**
## Summary

Remove `Box` from `Icon`.

Also removed the `title` property as it doesn't exist on `<svg>` elements

## Type

- Refactor

### CODE REVIEW

File: src/components/Icon/index.tsx [hunk 1]
```diff
@@ -435,13 +428,11 @@ const Icon = forwardRef<SVGElement, IconProps>(
 
     return (
       <StyledIcon
-        as="svg"
         ref={ref}
-        className="sc-ui-icon"
         color={color}
+        className="sc-ui-icon"
```
Comment (chambo-e) [inline src/components/Icon/index.tsx:433]: I think we can remove this class, it seems to no be used anymore

File: src/components/Icon/index.tsx [hunk 2]
```diff
@@ -406,19 +402,16 @@ export type IconName = keyof typeof ICONS
...
-const Icon = forwardRef<SVGElement, IconProps>(
+const Icon = forwardRef<SVGSVGElement, IconProps>(
   (
     {
       name = 'circle',
       color = 'currentColor',
       size = '1em',
-      verticalAlign = 'middle',
       prominence = 'default',
       ...props
```
Comment (chambo-e) [inline src/components/Icon/index.tsx:416]: Can we remove the spread ?

File: src/components/Icon/index.tsx [hunk 3]
```diff
@@ -406,19 +402,16 @@ export type IconName = keyof typeof ICONS
...
   name?: IconName
-  title?: string
   prominence?: ProminenceProps
   color?: Color | string
-} & BoxProps &
-  React.SVGAttributes<HTMLOrSVGElement>
+} & React.SVGAttributes<HTMLOrSVGElement>
 
-const Icon = forwardRef<SVGElement, IconProps>(
+const Icon = forwardRef<SVGSVGElement, IconProps>(
```
Comment (matthprost) [inline src/components/Icon/index.tsx:409]: What's `SVGSVGElement` ? Shouldn't it be `SVGElement` ?

File: src/components/Icon/index.tsx [hunk 4]
```diff
@@ -406,21 +402,20 @@ export type IconName = keyof typeof ICONS
 type IconProps = {
   size?: number | string
   name?: IconName
-  title?: string
   prominence?: ProminenceProps
   color?: Color | string
-} & BoxProps &
-  React.SVGAttributes<HTMLOrSVGElement>
+  'data-testid'?: string
+} & React.SVGAttributes<HTMLOrSVGElement>
```
Comment (chambo-e) [inline src/components/Icon/index.tsx:408]: This is not true anymore :)

Comment (baptisteArno) [reply src/components/Icon/index.tsx:408]: Oh yeah my bad 😮‍💨

**Review by matthprost — APPROVED** [review]

**Comment by QuiiBz** [comment]
Some type errors:

Screenshot 2022-10-14 at 16 10 40

**Review by QuiiBz — APPROVED** [review]

**Pull request merged**